code/workspaces/infrastructure-api/src/kubernetes/deploymentApi.spec.js

```diff
@@ -57,13 +57,14 @@ describe('Kubernetes Deployment API', () => {
         });
     });
 
-    it('should return an error if creation fails', () => {
+    it('should return an error if creation fails', async () => {
       mock.onPost(DEPLOYMENT_URL).reply(400, { message: 'error-message' });
 
-      return deploymentApi.createDeployment(DEPLOYMENT_NAME, NAMESPACE, manifest)
-        .catch((error) => {
-          expect(error.toString()).toEqual('Error: Unable to create kubernetes deployment error-message');
-        });
+      try {
+        await deploymentApi.createDeployment(DEPLOYMENT_NAME, NAMESPACE, manifest);
+      } catch (error) {
+        expect(error.toString()).toEqual('Error: Unable to create kubernetes deployment error-message');
+      }
```

Comment:
I don't know whether this still suffers from the same problem. I know in other places you have put `expect(false).toEqual(true)` to make sure there is a hard failure if the api call doesn't throw. 

Or in other places we have done the following ensuring the expect is always called.
```js
let error;
try {
  await apiCall();
} catch (err) {
  error = err;
}
expect(error).toEqual(...);
```